test/jsonapi/plugs/query_parser_test.exs

```diff
@@ -151,4 +153,23 @@ defmodule JSONAPI.QueryParserTest do
       get_view_for_type(MyView, "cupcake")
     end
   end
+
+  test "integrates with UnderscoreParameters to filter dasherized fields" do
+    # The incoming request has a dasherized filter name
+    conn =
+      :get
+      |> conn("?filter[favorite-food]=pizza")
+      |> put_req_header("content-type", JSONAPI.mime_type())
+
+    # The filter in the controller is expecting an underscored filter name
+    config = struct(Config, view: MyView, opts: [filter: ["favorite_food"]])
+
+    conn =
+      conn
+      |> JSONAPI.UnderscoreParameters.call(replace_query_params: true)
+      |> JSONAPI.QueryParser.call(config)
+
+    # Ensure the underscored file name is present in the parsed filters
+    assert [favorite_food: _] = conn.assigns.jsonapi_query.filter
+  end
```

Comment:
Annotation: This is the "main" test, which makes sure that the `UnderscoreParameters` and `QueryParser` plugs will work together.